src/components/Notifications/EventLog/EventLogTreeFilter.tsx

```diff
@@ -0,0 +1,214 @@
+
+    React.useEffect(() => {
+        if (groups.length !== 0) {
+            setTreeNodeById(produce((prev) => {
+                // In case Bundles request is pending during Tree View render
+                if (Object.keys(prev).length === 0) {
+                    return initialize;
+                }
+                else if (filters.length === 0) {
+                    return initialize;
+                }
+                else {
+                    filters.forEach(activeFilter => {
+                        const treeNode = prev[activeFilter.bundleId];
+                        const activeChips = activeFilter.chips.map(chip => chip.value);
+
+                        treeNode.children?.forEach(childNode => {
+                            childNode.checkProps.checked = activeChips.includes(childNode.id);
+                        });
+
+                        if (allChildrenChecked(treeNode)) {
+                            treeNode.checkProps.checked = true;
+                        }
+                        else if (childChecked(treeNode)) {
+                            treeNode.checkProps.checked = null;
+                        }
+                        else {
+                            treeNode.checkProps.checked = false;
+                        }
+                    });
+                }
+            }));
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [ groups, filters ]);
+
+    const flattenTree = React.useMemo(() => {
+        const flatTreeDataArray = produce(treeDataArray, (prev) => {
+            prev.forEach(treeNode => {
+                if (!!treeNode.children) {
+                    prev.push(...treeNode.children);
+                }
+            });
+        });
+
+        return flatTreeDataArray;
+    }, [ treeDataArray ]);
+
+    React.useEffect(() => {
+        const groupNames = groups.map(group => group.name);
+        const activeParentFilters = flattenTree.filter(treeNode => {
+            const isActive = treeNode.checkProps.checked;
+            return (!!isActive || isActive === null) && groupNames.includes(treeNode.id);
+        });
+
+        updateFilters(produce((prev) => {
+            if (prev.length === 0 && activeParentFilters.length === 0) {
+                return prev;
+            }
+
+            else if (prev.length !== 0 && activeParentFilters.length === 0) {
+                return [];
+            }
+
+            const newCustomFilters = activeParentFilters.map(parentFilter => ({
+                bundleId: parentFilter.id,
+                category: parentFilter.name as string,
+                chips: parentFilter.children?.filter(childNode => !!childNode.checkProps.checked).map(childFilter => ({
+                    name: childFilter.name as string,
+                    value: childFilter.id,
+                    isRead: true
+                })) ||
```

Comment:
Could be replaced with the [Nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)
In this case it's more or less the same, but might help when reading the code.
```suggestion
                })) ??
```

Reply:
Yeah, in this case, they would be doing the same thing... but null coalescing operator is more properly scoped since I only care about undefined cases in this snippet instead of all falsy values. Will change